dags/rlc.py

```diff
@@ -0,0 +1,335 @@
+            # Convert date strings into datetime objects for comparison
+            datetime_format_dates = []
+            for date_string in date_strings:
+                
+                try:
+                    datetime_date = parse(date_string, fuzzy=False)
+                    datetime_format_dates.append(datetime_date)
+                except ValueError:
+                    pass
+            
+            # Case 1: none of the strings in 'date_strings' can be converted to datetime objects
+            if not datetime_format_dates:
+                return None
+            # Case 2: only one date string can be converted into a datetime object
+            elif len(datetime_format_dates) == 1:
+                formatted_date = datetime_format_dates[0].strftime("%Y-%m-%d")
+                return formatted_date
+            # Case 3: more than one date string can be converted into datetime objects
+            else:
+                latest_date = datetime_format_dates[0]
+                for date in datetime_format_dates[1:]:
+                    if date > latest_date:
+                        latest_date = date
+                formatted_date = latest_date.strftime("%Y-%m-%d")
+                return formatted_date
+
+def pull_lpi(conn):
+    
+    local_table='vz_safety_programs_staging.signals_cart'
+    url = "https://secure.toronto.ca/opendata/cart/traffic_signals/v3?format=json"
+    return_json = requests.get(url).json()
+    
+    rows = []
+
+    # column names in the PG table
+    col_names = ['asset_type','px','main_street','midblock_route','side1_street','side2_street','latitude','longitude','details','activation_date'] 
+
+    # attribute names in JSON dict
+    att_names = ['px', 'main', 'mid_block', 'side1', 'side2', 'lat', 'long', 'lpiComment']
+
+    # each "info" is all the properties of one LPI, including its coords
+    for obj in return_json:
+        if obj['leading_pedestrian_intervals'] == 1:
+
+            # temporary list of properties of one LPI to be appended into the rows list
+            one_lpi = []
+
+            one_lpi.append('Leading Pedestrian Intervals') # append the asset_name as listed in EC2
+
+            # append the values in the same order as in the table
+            for attr in att_names:
+                one_lpi.append(obj[attr])
+
+            # append the latest LPI implementation date of any of the four directions
+            one_lpi.append(lastest_imp_date(obj))
+
+            rows.append(one_lpi)
+    
+    # delete existing LPI and insert into the local table
+    insert = 'INSERT INTO {} ({}) VALUES %s'.format(local_table, ','.join(col_names))
+    with conn:
+        with conn.cursor() as cur:
+            cur.execute("DELETE FROM {} WHERE asset_type = 'Leading Pedestrian Intervals'".format(local_table))
+            execute_values(cur, insert, rows)
+
+# ------------------------------------------------------------------------------
+# Set up the dag and task
+RLC_DAG = DAG(
```

Comment:
For consistency, it's better to change this variable to reflect the fact that it now represents multiple traffic signal tasks, not only RLC. In case of changing the name of this variable, make sure to update it in all other locations, e.g., while creating the tasks below